app/main.py

```diff
@@ -1,2 +1,62 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    class Node:
+        def __init__(self, key: Any, value: Any) -> None:
+            self.key = key
+            self.hash = hash(key)
+            self.value = value
+
+        def __repr__(self) -> str:
+            return f"({self.key}, {self.hash}, {self.value})"
+
+    def __init__(self,
+                 initial_capacity: int = 8,
+                 load_factor: float = 2 / 3) -> None:
+        self.capacity = initial_capacity
+        self.load_factor = load_factor
+        self.table = [None] * initial_capacity
+        self.size = 0
+
+    def index(self, key: Any) -> int:
+        return hash(key) % self.capacity
+
+    def resize(self) -> None:
+        old_table = self.table
+        self.capacity = self.capacity * 2
+        self.table = [None] * self.capacity
+        self.size = 0
```

Comment:
Resetting `self.size` to 0 in the `resize` method is not correct. The size should reflect the number of elements in the dictionary, which does not change during resizing. Consider removing this line.